content/extending-chat-widget/custom-identity-provider/index.mdx

```diff
@@ -0,0 +1,173 @@
+
+## Setup
+Customer Identity Provider is essentially a singleton function that you can define as a part of `LiveChatWidget` object in tracking code that you use to install ChatWidget on your website.
+
+```html
+<!-- Start of LiveChat (www.livechat.com) code -->
+<script>
+    window.__lc = window.__lc || {};
+    window.__lc.license = <LICENSE_NUMBER>;
+    window.__lc.custom_identity_provider = function() {
+      return {
+          getToken,
+          getFreshToken,
+          hasToken,
+          invalidate
+      }
+    }
+    ;(function(n,t,c){function i(n){return e._h?e._h.apply(null,n):e._q.push(n)}var e={_q:[],_h:null,_v:"2.0",on:function(){i(["on",c.call(arguments)])},once:function(){i(["once",c.call(arguments)])},off:function(){i(["off",c.call(arguments)])},get:function(){if(!e._h)throw new Error("[LiveChatWidget] You can't use getters before load.");return i(["get",c.call(arguments)])},call:function(){i(["call",c.call(arguments)])},init:function(){var n=t.createElement("script");n.async=!0,n.type="text/javascript",n.src="https://cdn.livechatinc.com/tracking.js",t.head.appendChild(n)}};!n.__lc.asyncInit&&e.init(),n.LiveChatWidget=n.LiveChatWidget||e}(window,document,[].slice))
+</script>
+<noscript><a href="https://www.livechat.com/chat-with/<LICENSE_NUMBER>/" rel="nofollow">Chat with us</a>, powered by <a href="https://www.livechat.com/?welcome" rel="noopener nofollow" target="_blank">LiveChat</a></noscript>
+<!-- End of LiveChat code -->
+```
+
+# Methods
+
+The aforementioned function assigned to `custom_identity_provider` property should return a set of the following promises:
+
+- `getToken` - resolving [Chat Widget token](#chat-widget-token). If you want to cache the token in some way, this should return cached token instead of fresh request to API.
+- `getFreshToken` - resolving [Chat Widget token](#chat-widget-token). This should always make a call for fresh token from accounts API.
+- `hasToken` - resolving boolean. Determining whether token has been acquired.
+- `invalidate` - resolving nothing. When called should remove current token. There is no need to do anything else as new token will be requested by `getFreshToken` afterwards.
+
+# Chat Widget Token
+
+LiveChat authorization token you are willing to resolve in functions should contain the following properties:
+
+| Parameter      | Description                                                                      |
+| -------------- | -------------------------------------------------------------------------------- |
+| `accessToken`  | A token you can use to call LiveChat APIs on behalf of the user (customer).      |
+| `entityId`     | The ID of customer.                                                              |
+| `expiresIn`    | A number in **milliseconds** specifying how long the `accessToken` will be valid.|
+| `tokenType`    | Value: `Bearer`.                                                                 |
```

Comment:
@klarzynskik in case of a shape that describes the token object in the Chat Widget then yes it will always be an Bearer tokenType, this shape is not describing a direct output from accounts.